core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/PauseManager.scala

```diff
@@ -1,43 +1,32 @@
 package edu.uci.ics.amber.engine.architecture.worker
 
-object PauseManager {
-  final val NoPause = 0
-  final val Paused = 1
+import scala.collection.mutable
 
+object PauseManager {
   final case class ExecutionPaused()
 }
 
 class PauseManager {
 
-  // current pause privilege level
-  private var pausePrivilegeLevel = PauseManager.NoPause
-
-  // used temporarily for reshape in sort.
-  // TODO: Refactor to expose various pause levels
-  var pausedByOperatorLogic = false
-
-  /** pause functionality
-    * both dp thread and actor can call this function
-    * @param
-    */
-  def pause(): Unit = {
-
-    /*this line atomically applies the following logic:
-      Level = Paused
-      if(level >= pausePrivilegeLevel.get())
-        pausePrivilegeLevel.set(level)
-     */
-    pausePrivilegeLevel = PauseManager.Paused
+  private val pauseInvocations = new mutable.HashMap[PauseType.Value, Boolean]()
+
+  def recordRequest(pauseType: PauseType.Value, enablePause: Boolean): Unit = {
+    pauseInvocations(pauseType) = enablePause
   }
 
-  def isPaused: Boolean = pausePrivilegeLevel == PauseManager.Paused
+  def getPauseStatusByType(pauseType: PauseType.Value): Boolean =
+    pauseInvocations.getOrElse(pauseType, false)
+
+  def isPaused(): Boolean = {
+    var isPaused = false
+    pauseInvocations.foreach(entry => {
+      if (entry._2) { isPaused = true }
+    })
+    isPaused
+  }
 
-  /** resume functionality
-    * only actor calls this function for now
-    * @param
-    */
-  def resume(): Unit = {
-    pausePrivilegeLevel = PauseManager.NoPause
+  def canEnableDataQueue(): Boolean = {
+    pauseInvocations.forall(entry => entry._2 == false)
```

Comment:
I think this function is just `! isPaused()`, maybe removed it?

Reply:
agreed